src/server/generic_family.cc

```diff
@@ -597,6 +597,7 @@ void OpScan(const OpArgs& op_args, const ScanOpts& scan_opts, uint64_t* cursor,
   auto& db_slice = op_args.GetDbSlice();
   DCHECK(db_slice.IsDbValid(op_args.db_cntx.db_index));
 
+  db_slice.BlockingCounter()->Wait();
```

Comment:
this requires a comment
+add a fiber gourd here to make sure we dont preemt later in this flow otherwise this wait does not holds